### DISCUSSION TITLE
Why is column A not indexed?

### DISCUSSION THREAD

**Discussion by assertnotnull** [prisma/prisma#19429] [Q&A] [answered]
Why is column A not indexed in a M2M table ?
As per the [doc](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) says:
```
CREATE TABLE "_CategoryToPost" (
    "A" integer NOT NULL REFERENCES "Category"(id) ,
    "B" integer NOT NULL REFERENCES "Post"(id)
);
CREATE UNIQUE INDEX "_CategoryToPost_AB_unique" ON "_CategoryToPost"("A" int4_ops,"B" int4_ops);
CREATE INDEX "_CategoryToPost_B_index" ON "_CategoryToPost"("B" int4_ops);

CREATE TABLE "Category" (
    id integer SERIAL PRIMARY KEY
);

CREATE TABLE "Post" (
    id integer SERIAL PRIMARY KEY
);
```

From what I see it means that filtering for the values of A is always slower than filtering for values of B and that performance is just dependent on how you named your table (since the name of the join table is created by alphabetical property).

As result we have to manually create those indexes and they get removed every time we do a migration.

**Comment by janpio** [comment]
An index on `A-B` has the same effect than one on just `A`. 
A plain `A` one might be preferred when you really only need one for `A`, but then on insert that affect `A` you would always have to update both indexes - for no good reason.

Or are you seeing that index not being used in your case when you filter/sort by `A`?

**Reply by assertnotnull** [reply]
We saw an increase in performance after we added an index on `A`

**Reply by janpio** [reply]
On read or on writes?
Did you measure the impact or analyze it with an `EXPLAIN` of the query?

**Reply by assertnotnull** [reply, marked as answer]
It was on read. We didn't measure it but we saw an increase of performance on our web interface after this change.

**Reply by assertnotnull** [reply]
It seems we don't know in the end as it was not measured..
